docs/overrides/main.html

```diff
@@ -3,5 +3,90 @@
 {% block content %}
 {{ super() }}
 <script src="/flarelytics/client.js"></script>
+<script>
+    if (!window.__posthogInitialized) {
+
+        // Initialize PostHog
+        !function(t,e){var o,n,p,r;e.__SV||(window.posthog=e,e._i=[],e.init=function(i,s,a){function g(t,e){var o=e.split(".");2==o.length&&(t=t[o[0]],e=o[1]),t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}}(p=t.createElement("script")).type="text/javascript",p.crossOrigin="anonymous",p.async=!0,p.src=s.api_host.replace(".i.posthog.com","-assets.i.posthog.com")+"/static/array.js",(r=t.getElementsByTagName("script")[0]).parentNode.insertBefore(p,r);var u=e;for(void 0!==a?u=e[a]=[]:a="posthog",u.people=u.people||[],u.toString=function(t){var e="posthog";return"posthog"!==a&&(e+="."+a),t||(e+=" (stub)"),e},u.people.toString=function(){return u.toString(1)+".people (stub)"},o="init capture register register_once register_for_session unregister unregister_for_session getFeatureFlag getFeatureFlagPayload isFeatureEnabled reloadFeatureFlags updateEarlyAccessFeatureEnrollment getEarlyAccessFeatures on onFeatureFlags onSessionId getSurveys getActiveMatchingSurveys renderSurvey canRenderSurvey getNextSurveyStep identify setPersonProperties group resetGroups setPersonPropertiesForFlags resetPersonPropertiesForFlags setGroupPropertiesForFlags resetGroupPropertiesForFlags reset get_distinct_id getGroups get_session_id get_session_replay_url alias set_config startSessionRecording stopSessionRecording sessionRecordingStarted captureException loadToolbar get_property getSessionProperty createPersonProfile opt_in_capturing opt_out_capturing has_opted_in_capturing has_opted_out_capturing clear_opt_in_out_capturing debug".split(" "),n=0;n<o.length;n++)g(u,o[n]);e._i.push([i,s,a])},e.__SV=1)}(document,window.posthog||[]);
+
+        try {
+            posthog.init('phc_WEVBsfcilPS9u6e0CY22d8DqtFt0zalAP9A6iyCBHNH', {
+                api_host: 'https://eu.i.posthog.com',
+                persistence: 'memory'
+            });
+            window.__posthogInitialized = true;
+        } catch (error) {
+            console.error('PostHog initialization failed:', error);
+        }}
+
+document.addEventListener('DOMContentLoaded', function() {
+
+    // Setup
+    const elements = {
+        form: document.querySelector('.md-search__form'),
+        input: document.querySelector('[data-md-component="search-query"]'),
+        results: document.querySelector('.md-search__output')
+    };
+
+    if (!elements.form || !elements.input || !elements.results) return;
+
+    let searchTimeout;
+    let lastSearch = { term: '', time: 0 };
+
+    // Track search with debouncing
+    function trackSearch(term, type) {
+        if (term.length < 3 || term === lastSearch.term || Date.now() - lastSearch.time < 1000) return;
+
+        setTimeout(() => {
```

Comment:
please add a comment explaining why setTimeout is used here. i'm guessing it's to give mkdocs time to render search results? is 100ms reliably enough?

Reply:
Added, also bumped to 200ms which I reckon is enough